### DISCUSSION TITLE
cpg0028-kelley-resistance

### DISCUSSION THREAD

**Discussion by shntnu** [broadinstitute/cellpainting-gallery#55] [Dataset descriptions]
Imaging Platform project tag = 2018_05_30_ResistanceMechanisms_Kapoor
Segmentation/ Feature extraction is being performed by Cimini lab (done)
Profile creation is being performed by Carpenter-Singh lab (done)
Data can be public in RODA Immediately

Update as generated:  

Link to profile repo (same as publication repo): https://github.com/broadinstitute/profiling-resistance-mechanisms/tree/master/0.generate-profiles/profiles
Link to publication repo: https://github.com/broadinstitute/profiling-resistance-mechanisms
cellpainting-gallery identifier: `cpg0028-kelley-resistance`

- [ ] Metadata collection form filled out by collaborator or Imaging Platform member (https://airtable.com/shrVxz9DcoMlDoCBI)
- [ ] Metadata completely filled out in Project Profiler Database (Imaging Platform internal use only)
- [x] Segmentation/Feature extraction complete
- [x] Profiling complete

Transfer to CellPainting Gallery:
- [ ] Upload data to RODA (is private by default)
- [ ] Run validation script to ensure completion (https://github.com/jump-cellpainting/data-validation)
- [ ] Update cellpainting-gallery/README.md
- [ ] Make RODA entry public

If data is being published, prepare for publication:
- [ ] Run Distributed-BioFormats2Raw to create .ome.zarr files
- [ ] Upload (meta)data to IDR (images remain hosted in cellpainting-gallery).

Once published:
- [ ] Make IDR entry public
- [ ] Update cellpainting-gallery/README.md and open-data-registry/cellpainting-gallery.yml to reflect publication
- [ ] Move this Issue from cellpainting-gallery-private to cellpainting-gallery. This step can be performed at an earlier point if it needs inputs from an external collaborator.

**Comment by shntnu** [comment]
- [x] make `s3://imaging-platform/projects/2018_05_30_ResistanceMechanisms_Kapoor/` publicly accessible (for now) to simplify the process
- [x] run `python restore_intelligent.py imaging-platform projects/2018_05_30_ResistanceMechanisms_Kapoor` to thaw ("restore") the files
- [x] Create issue in https://github.com/broadinstitute/cellpainting-gallery/issues/

**Comment by shntnu** [comment]
@gwaybio 

The final folder structure should look like this https://github.com/broadinstitute/cellpainting-gallery/blob/main/folder_structure.md

We need someone to provide the `aws s3 sync` instructions to copy the files over to s3://cellpainting-gallery. I've started a stub here https://github.com/broadinstitute/profiling-resistance-mechanisms/blob/cpg-upload/scripts/cpg_upload.sh (borrowing from https://github.com/broadinstitute/cellpainting-gallery/blob/main/upload.md#uploading-multiple-plates-at-once)

We can tag team between you, me, @mekelley and @yhan8 to get this done.

Any thoughts on how to get going on this?

**Comment by gwaybio** [comment]
Thanks @shntnu - just now seeing this post.

I don't know how we could get going on this. Is this something you can handle (maybe looping in Yu)? We could get Dave on this, but I fear it would only take you longer to explain than to do

**Comment by ErinWeisbart** [comment]
I can transfer files to cpg. @gwaybio can you clarify  - should I transfer all data or only batches larger than a certain number?
For reference, for other datasets in the cellpainting-gallery we are including all data (including pilots) even if they are not in the final publication.

**Comment by shntnu** [comment]
Very grateful, @ErinWeisbart !!

> should I transfer all data or only batches larger than a certain number?
> For reference, for other datasets in the cellpainting-gallery we are including all data (including pilots) even if they are not in the final publication.

Megan had some notes here https://github.com/broadinstitute/profiling-resistance-mechanisms/issues/122#issuecomment-1287279400 where she linked to a spreadsheet indicating the batches to upload. 

I like the idea of uploading everything.

Do you need any more information, @ErinWeisbart ?

**Comment by gwaybio** [comment]
Thank you @ErinWeisbart !! 🙏

**Comment by ErinWeisbart** [comment]
I will go ahead and upload all, since it sounds like nobody objects. Don't think I need more info but will re-ping if I hit any stumbling blocks.

**Comment by ErinWeisbart** [comment]
One question for @gwaybio (and/or @shntnu) - should the `workspace/parameters` folder be synced?
I'm not sure what it is. It looks to be complete and consistent across all batches and is not large.

**Reply by gwaybio** [reply]
I also do not know what this is! My guess is that since it is small, it is ok to add. Is there a README describing the contents?

**Reply by shntnu** [reply]
That stored params for feature selection using the https://github.com/cytomining/cytotools workflows, now deprecated with profiling-recipe

Please do sync it and I can review and decide if it should be deleted

**Reply by ErinWeisbart** [reply]
`parameters` are now synced.

**Reply by shntnu** [reply]
It's ok to keep these parameters, although there's a very low probability of ever using them. Here's what's in them, using an example from `cpg0028-kelley-resistance/broad/workspace/parameters/2021_03_03_Batch12`

```
2023-11-02 10:42:02    2059960 sample/2021_03_03_Batch12_augmented_sample.feather
2023-11-02 10:42:02    2070040 sample/2021_03_03_Batch12_normalized_sample.feather
2023-11-02 10:42:02       5589 variable_selection/correlation_threshold.txt
2023-11-02 10:42:02     126861 variable_selection/manual.txt
```

The `*_sample.feather` are subsamples of the profiles on which the feature selection is performed.

The text files were created using https://github.com/cytomining/cytotools/blob/master/R/preselect.R and are intermediate files generated to feature select.

```
aws s3 cp s3://cellpainting-gallery/cpg0028-kelley-resistance/broad/workspace/parameters/2021_03_03_Batch12/variable_selection/correlation_threshold.txt -|head
variable
Cells_AreaShape_Compactness
Cells_AreaShape_EulerNumber
Cells_AreaShape_Extent
Cells_AreaShape_Orientation
Cells_AreaShape_Zernike_0_0
Cells_AreaShape_Zernike_1_1
Cells_AreaShape_Zernike_2_0
Cells_AreaShape_Zernike_2_2
Cells_AreaShape_Zernike_3_1
```

```
aws s3 cp s3://cellpainting-gallery/cpg0028-kelley-resistance/broad/workspace/parameters/2021_03_03_Batch12/variable_selection/manual.txt -|head
variable
Cells_AreaShape_Area
Cells_AreaShape_Center_X
Cells_AreaShape_Center_Y
Cells_AreaShape_Center_Z
Cells_AreaShape_Compactness
Cells_AreaShape_Eccentricity
Cells_AreaShape_EulerNumber
Cells_AreaShape_Extent
```

**Comment by GaiaGerbaka** [comment]
In the context of harmonizing the CPG metadata, I noticed that the Cell Painting protocol version was missing from the README on the main page for this project. The publication references the [Bray et al., 2016](https://elifesciences.org/articles/91362#bib3) paper but this reference does not specify the CP version. It would be helpful if someone could provide this information, along with the modality used for imaging (confocal or widefield).

**Reply by ErinWeisbart** [reply]
Bray et al is v2. See [Cell Painting wiki](https://github.com/carpenter-singh-lab/2023_Cimini_NatureProtocols/wiki#changes-in-the-official-protocol-to-create-v2-bray-et-al-2016).
The images are widefield.

**Comment by ErinWeisbart** [comment]
@GaiaGerbaka made load_data.csv's for this prefix (as we were not able to find the original).
Code she used for load_data.csv creation is at https://github.com/broadinstitute/GaiaGerbaka_Projects/blob/main/Metadata_Project/data_scraping_cpg0028.ipynb
I uploaded them today.
